database/database.go

```diff
@@ -6,61 +6,57 @@ import (
 	GetCourse(uint64, bool) (*pb.Course, error)
-	GetCourseByOrganizationID(did uint64) (*pb.Course, error)
+	// GetCourseByOrganizationID fetches course by organization ID.
+	GetCourseByOrganizationID(organizationID uint64) (*pb.Course, error)
+	// GetCourses returns a list of courses. If one or more course IDs are provided,
+	// the corresponding courses are returned. Otherwise, all courses are returned.
 	GetCourses(...uint64) ([]*pb.Course, error)
-	GetCoursesByUser(uid uint64, statuses ...pb.Enrollment_UserStatus) ([]*pb.Course, error)
+	// GetCoursesByUser returns all courses (with enrollment status)
+	// for the given user id.
+	// If enrollment statuses is provided, the set of courses returned
+	// is filtered according to these enrollment statuses.
+	GetCoursesByUser(userID uint64, statuses ...pb.Enrollment_UserStatus) ([]*pb.Course, error)
+	// UpdateCourse updates course information.
 	UpdateCourse(*pb.Course) error
 
+	// CreateEnrollment creates a new pending enrollment.
 	CreateEnrollment(*pb.Enrollment) error
-	RejectEnrollment(uid uint64, cid uint64) error
-	EnrollStudent(uid uint64, cid uint64) error
-	EnrollTeacher(uid uint64, cid uint64) error
-	SetPendingEnrollment(uid, cid uint64) error
+	// RejectEnrollment removes the user enrollment from the database
+	RejectEnrollment(userID, courseID uint64) error
+	// UpdateEnrollmentStatus changes status of the course enrollment for the given user and course.
+	UpdateEnrollmentStatus(userID, courseID uint64, status pb.Enrollment_UserStatus) error
 	// UpdateGroupEnrollment is used to reset group ID when previously aproved group is
 	// being removed or a user is removed from the group
-	UpdateGroupEnrollment(uid, cid uint64) error
-
-	GetEnrollmentsByCourse(cid uint64, statuses ...pb.Enrollment_UserStatus) ([]*pb.Enrollment, error)
-	GetEnrollmentByCourseAndUser(cid uint64, uid uint64) (*pb.Enrollment, error)
-	// CreateAssignment creates a new or updates an existing assignment.
-	CreateAssignment(*pb.Assignment) error
-	// UpdateAssignments updates the specified list of assignments.
-	UpdateAssignments([]*pb.Assignment) error
-	GetAssignmentsByCourse(uint64) ([]*pb.Assignment, error)
-	GetNextAssignment(cid, uid, gid uint64) (*pb.Assignment, error)
-	GetAssignment(query *pb.Assignment) (*pb.Assignment, error)
-
-	// CreateSubmission creates a submission in the database.
-	CreateSubmission(*pb.Submission) error
-	// UpdateSubmission updates the specified submission with approved or not approved.
-	UpdateSubmission(submissionID uint64, approved bool) error
-	// GetSubmission returns a single submission matching the given query.
-	GetSubmission(query *pb.Submission) (*pb.Submission, error)
-	// GetSubmissions returns a list of submission entries for the given course, matching the given query.
-	GetSubmissions(cid uint64, query *pb.Submission) ([]*pb.Submission, error)
-	GetCourseSubmissions(uint64, bool) ([]pb.Submission, error)
+	UpdateGroupEnrollment(userID, courseID uint64) error
```

Comment:
The func UpdateGroupEnrollment is not used in the code (except for one test). Is this still useful to keep? Also the comment has a typo "aproved" and is also confusing that we don't take a groupID or something as input here...
The comment in `gorm.db` says: _UpdateGroupEnrollment sets GroupID of a student enrollment to 0._

Reply:
Hm, it is supposed to be used, at least the test starts failing when this part is removed. I'll re-check.